providers/implementations/kem/template_kem.c

```diff
@@ -156,17 +154,15 @@ static int template_encapsulate(void *vctx, unsigned char *out, size_t *outlen,
 static int template_decapsulate(void *vctx, unsigned char *out, size_t *outlen,
                                 const unsigned char *in, size_t inlen)
 {
-    debug_print("decaps %p to %p inlen at %d\n", vctx, out, inlen);
+    debug_print("decaps %p to %p inlen at %u\n", vctx, out, inlen);
 
     /* add algorithm-specific length checks */
 
     if (outlen != NULL)
         *outlen = 0; /* replace with shared secret length */
 
-    if (out == NULL) {
-        debug_print("decaps outlen set to %d \n", *outlen);
+    if (out == NULL)
```

Comment:
Same here -- the template cannot put a different value than 0 (or 42 if you like :) but in real code the true value drives a lot of logic within `openssl` core -- and getting it wrong causes harm; hence the proposal to make it visible.